### DISCUSSION TITLE
JSON_LISTINGS_FEED Help with "Invalid empty value provided in patch at index of 0" Error

### DISCUSSION THREAD

**Discussion by cnr91** [amzn/selling-partner-api-models#4066] [General] [answered]
Hello,

We are encountering a persistent issue while using Amazon SP-API with JSON_LISTINGS_FEED to add an existing ASIN to our inventory. We keep receiving the error "Invalid empty value provided in patch at index of 0". Below is the JSON we are sending:

`{
  "header": {
    "sellerId": "HIDDEN",
    "version": "2.0",
    "issueLocale": "en_US"
  },
  "messages": [
    {
      "messageId": 1,
      "sku": "ABC123",
      "operationType": "PARTIAL_UPDATE",
      "productType": "PRODUCT",
      "attributes": {
        "externalProductId": "B007KQBXN0",
        "externalProductIdType": "ASIN",
        "fulfillment_availability": [
          {
            "fulfillment_channel_code": "DEFAULT",
            "quantity": 10
          }
        ],
        "standard_price": {
          "currency": "USD",
          "amount": 99.99
        }
      }
    }
  ]
}
`

Error Detail:
`{
  "header": {
    "sellerId": "HIDDEN",
    "version": "2.0",
    "feedId": "103771019934"
  },
  "issues": [
    {
      "messageId": 1,
      "code": "BAD_REQUEST",
      "severity": "ERROR",
      "message": "Invalid empty value provided in patch at index of 0."
    }
  ],
  "summary": {
    "errors": 1,
    "warnings": 0,
    "messagesProcessed": 1,
    "messagesAccepted": 0,
    "messagesInvalid": 1
  }
}
`

**Comment by weilinggu** [comment, marked as answer]
@cnr91 What is the product type of the product? Have you validated your attributes against the product type JSON schema? Are you trying to create an offer for an existing product?
Just from a glance, it seems like perhaps you are not using the right attribute or providing the attribute data in correct format.

You can also try using the [Listings sample application](https://github.com/amzn/selling-partner-api-samples/tree/main/use-cases/listings) to get an example of JSON feed you can use. It has a feature to generate JSON feed based on the operation you are trying to do. There is also [a series of walk through videos](https://www.youtube.com/watch?v=JEZk2Id5j2s&list=PLUFIeZA9F5Px4FbEU_VJf7Ud3azk9YMEt&index=7) on how to use it.

**Comment by chapmanjw** [comment]
The attributes provided (`externalProductId`, `standard_price`, etc.) are all malformed. The Product Type Definitions API provides JSON Schemas describing the proper format for attributes used in the JSON_LISTINGS_FEED format: https://developer-docs.amazon.com/sp-api/docs/product-type-api-use-case-guide.

**Comment by cnr91** [comment]
First of all, thank you for your reply.
I didn't know there was a test software for SP-API, I was sending requests via Postman.
I can add and delete products with asin via POST_FLAT_FILE_INVLOADER_DATA, but as far as I understand, January 2025 will be completely closed on this side. So it made more sense to do these operations with JSON_LISTINGS_FEED.

[Listings sample application](https://github.com/amzn/selling-partner-api-samples/tree/main/use-cases/listings)
I made all the settings from the link you provided but it doesn't accept my information.

Settings:
![image](https://github.com/user-attachments/assets/528746fa-e492-4acd-b2bd-9eb8f7a082f0)
Error:
<img width="387" alt="image" src="https://github.com/user-attachments/assets/ce021a35-31d9-4f29-989f-2305604a192d">
![image](https://github.com/user-attachments/assets/530d414b-61fb-4eda-95f8-f37529c2f697)

At the moment, I have not yet finished all the suggestions you have given, but I will try all of them

**Reply by chapmanjw** [reply]
@cnr91 For the issues you're facing with the sample application, can you open a support ticket with developer support? https://developer.amazonservices.com/support. This is one of our newer offerings and we have a bug there to track down.

**Reply by cnr91** [reply]
> @cnr91 For the issues you're facing with the sample application, can you open a support ticket with developer support? https://developer.amazonservices.com/support. This is one of our newer offerings and we have a bug there to track down.

I had 3 different profiles in the aws cli setting, so I got an error.
`aws configure list-profiles`

Thank you for your quick response.

**Reply by weilinggu** [reply]
@cnr91 Is your issue resolved? or Are you still having issues?

**Comment by LeiSurrre** [comment]
Hi, I am also getting `Invalid empty value provided in patch at index of 0`
I was trying to migrate from POST_FLAT_FILE_PRICEANDQUANTITYONLY_UPDATE_DATA and what I had in the xml was just sku and quantity, and I only want a bulk update quantity. but when I try the json one, I have no idea where the productType is from, and I dont remember I set it up anywhere.

**Reply by LeiSurrre** [reply]
the json i was sending through the api and it is valid when i validate it against the schema
`{
    "header": {
        "sellerId": "A2F0GYYJN8NJN3",
        "version": "2.0",
        "issueLocale": "en_US"
    },
    "messages":[
        {
    "messageId": 1,
    "operationType": "PARTIAL_UPDATE",
    "sku": "inFlow-kit",
    "productType": "WIRELESS_ACCESSORY",
    "attributes": {
      "quantity": 0
    }
}
    ]
}`
